src/validators/SignedKeyRequestValidator.sol

```diff
@@ -0,0 +1,136 @@
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Emit an event when the admin sets a new IdRegistry contract address.
+     *
+     * @param oldIdRegistry The previous IdRegistry address.
+     * @param newIdRegistry The ne IdRegistry address.
+     */
+    event SetIdRegistry(address oldIdRegistry, address newIdRegistry);
+
+    /*//////////////////////////////////////////////////////////////
+                              CONSTANTS
+    //////////////////////////////////////////////////////////////*/
+
+    bytes32 internal constant _METADATA_TYPEHASH =
+        keccak256("SignedKeyRequest(uint256 userFid,uint256 requestingFid,bytes signerPubKey)");
+
+    /*//////////////////////////////////////////////////////////////
+                                STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev The IdRegistry contract.
+     */
+    IdRegistryLike public idRegistry;
+
+    /*//////////////////////////////////////////////////////////////
+                               CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Set the IdRegistry and owner.
+     *
+     * @param _idRegistry   IdRegistry contract address.
+     * @param _initialOwner Initial contract owner address.
+     */
+    constructor(address _idRegistry, address _initialOwner) EIP712("Farcaster MetadataValidator", "1") {
+        idRegistry = IdRegistryLike(_idRegistry);
+        _transferOwnership(_initialOwner);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                               VALIDATION
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Validate the SignedKeyRequest metadata associated with a signer key.
+     *         (Key type 1, Metadata type 1)
+     *
+     * @param userFid               The fid of the end user adding this signer key.
+     * @param signerPubKey          The public key of the signer.
+     * @param signedKeyRequestBytes An abi-encoded SignedKeyRequest struct, provided as the
+     *                              metadata argument to KeyRegistry.add.
+     *
+     * @return true if signature is valid and signer owns appFid, false otherwise.
+     */
+    function validate(
+        uint256 userFid,
+        bytes memory signerPubKey,
+        bytes calldata signedKeyRequestBytes
+    ) external view returns (bool) {
+        SignedKeyRequest memory signedKeyRequest = abi.decode(signedKeyRequestBytes, (SignedKeyRequest));
+
+        if (idRegistry.idOf(signedKeyRequest.requestSigner) != signedKeyRequest.requestingFid) return false;
+
+        /**
+         *  Safety: Since keys may only be registered once, they are not
+         *  vulnerable to replay, so we omit nonce and deadline in the
+         *  validation signature.
+         */
```

Comment:
```suggestion
        /**
         *  Safety: Since keys may only be registered once, request
         *  signatures are not vulnerable to replay, so we omit 
         *  nonce and deadline in the validation signature.
         */
```